data/scripts/import_acs_data.py

```diff
@@ -0,0 +1,111 @@
+import os
+import sys
+import csv
+
+import yaml
+from census import Census
+
+# Relevant FIPS codes for geographic queries
+US_FIPS = '1'
+PA_FIPS = '42'
+PHILLY_FIPS = '101'
+
+
+class ACSWriter(object):
+    """
+    Thin wrapper around a csv.DictWriter to perform some common formatting
+    for ACS API responses.
+    """
+    def __init__(self, output, variables):
+        """
+        - output: A file-like object to write the CSV to
+        - variables: A dict of ACS variables parsed from a .config file
+                     (see variables.yml for an example)
+        """
+        self.codes_to_names = {var['code']: var['name'] for var in variables}
+        fieldnames = ['fips'] + [var['name'] for var in variables]
+        self.writer = csv.DictWriter(output, fieldnames=fieldnames)
+
+    def writeheader(self):
+        self.writer.writeheader()
+
+    def write_acs_row(self, fips, acs_row):
+        row = {'fips': fips}
+        for var_code, var_stat in acs_row.items():
+            # Skip fields that don't match queried variables
+            if var_code not in self.codes_to_names.keys():
+                continue
+            # Process nulls
+            if var_stat == -666666666:
+                var_stat = None
+
+            row[self.codes_to_names[var_code]] = var_stat
+
+        self.writer.writerow(row)
+
+
+if __name__ == '__main__':
+    # Parse ACS variables from the config file
+    with open('./variables.yml') as variables_file:
+        variables_yml = yaml.load(variables_file)
+
+    # Parse variable argument from the command line
+    try:
+        census_var = sys.argv[1]
+    except IndexError:
+        raise NameError('import_acs_data.py requires an ACS variable argument')
+
+    if census_var not in variables_yml.keys():
+        msg = (f'Variable {census_var} not valid, must be one of:'
+               f' {list(variables_yml.keys())}')
+        raise NameError(msg)
+
+    variables = variables_yml[census_var]['variables']
+    codes = [var['code'] for var in variables]
+
+    # Parse year from the command line
+    try:
+        year = sys.argv[2]
+    except IndexError:
+        raise NameError('import_acs_data.py requires a year argument')
+
+    year = int(year)
+
+    # Init CSV writer object
+    writer = ACSWriter(sys.stdout, variables)
+    writer.writeheader()
+
+    # Init Census API object
+    c = Census(os.environ['CENSUS_API_KEY'], year=year)
+
+    # National-level data
+    us_res = c.acs5.us(codes)
+    assert len(us_res) > 0
+    for row in us_res:
+        writer.write_acs_row(US_FIPS, row)
+
+    # State-level data
+    pa_res = c.acs5.state(codes, PA_FIPS)
+    assert len(pa_res) > 0
+    for row in pa_res:
+        writer.write_acs_row(PA_FIPS, row)
+
+    # County-level data
+    philly_res = c.acs5.state_county(codes, PA_FIPS, PHILLY_FIPS)
```

Comment:
why is philadelphia county the right unit, and not the city of philadelphia?

If philadelphia county is the right unti, I would call this philly_county_fips.

Reply:
Is there an ACS geography that corresponds to municipalities like Philly? I didn't immediately see one in the `census` docs. I went with the county because the county and city are coterminous: https://en.wikipedia.org/wiki/Philadelphia_County,_Pennsylvania

Reply:
Named places. This was one of the reasons why we originally wrote https://github.com/datamade/census_area/#usage

Newer versions of the Census API actually have native support for named places, which we don't have convenience methods for in our census wrapper. (https://github.com/uscensusbureau/api-geoHierarchy-changes/blob/master/changes.md?eml=gd&utm_medium=email&utm_source=govdelivery)

Since they are coterminous, this seems fine. It would be good to add a comment about that fact.